graphix/pattern.py

```diff
@@ -1416,12 +1416,46 @@ def run_pattern(self, backend, **kwargs):
         result = exe.run()
         return result
 
-    def perform_pauli_measurements(self, leave_input=False, use_rustworkx=False):
+    def perform_pauli_measurements(
+        self, leave_input: bool = False, use_rustworkx: bool = False, ignore_pauli_with_deps: bool = False
+    ) -> None:
         """Perform Pauli measurements in the pattern using efficient stabilizer simulator.
 
+        Parameters
+        ----------
+        leave_input : bool
+            Optional (`False` by default).
+            If `True`, measurements on input nodes are preserved as-is in the pattern.
+        use_rustworkx : bool
+            Optional (`False` by default).
+            If `True`, `rustworkx` is used for fast graph processing.
+            If `False`, `networkx` is used.
+        ignore_pauli_with_deps : bool
+            Optional (`False` by default).
+            If `True`, Pauli measurements with domains depending on other measures are preserved as-is in the pattern.
+            If `False`, all Pauli measurements are preprocessed. Formally, measurements are swapped so that all Pauli measurements are applied first, and domains are updated accordingly.
+
         .. seealso:: :func:`measure_pauli`
 
         """
+        if not ignore_pauli_with_deps:
+            self.standardize()
+            pauli_nodes = self.extract_pauli_nodes()
+            # Create a new sequence with all Pauli nodes to the front
+            new_seq = []
+            pauli_nodes_inserted = False
+            for cmd in self:
+                if cmd.kind == CommandKind.M:
+                    if cmd.node not in pauli_nodes:
+                        if not pauli_nodes_inserted:
+                            new_seq.extend(pauli_nodes.values())
+                            pauli_nodes_inserted = True
+                        new_seq.append(cmd)
+                else:
+                    new_seq.append(cmd)
+            if not pauli_nodes_inserted:
+                new_seq.extend(pauli_nodes.values())
```

Comment:
How is this possible? I guess you need a standardization (I know we do it in `measure_pauli` but we should be explicit rather than implicit.

Reply:
Also, this might move the comand across possible dependency. please add a line of two of notes about  consequences of this skipping or if this doesn't affect at all, then why so.

Reply:
I changed to a clearer structure in 7483827, providing proofs of the signal shifting: the function `extract_pauli_nodes` has been renamed to `move_pauli_measurements_to_the_front` and is now responsible for both shifting and moving the Pauli measurements in the sequence.